### ISSUE TITLE
"Days" keyword in INSERT statements

### ISSUE THREAD

**Issue by mmiddleton3301** [antlr/grammars-v4#754]
I've been using the TSQL grammar in a project recently - firstly, let me say, fantastic work, and thank you to all the people who have contributed to this grammar! 😄 Apologies if I don't provide enough detail or provide it in the wrong format; please feel free to get back to me if you require more from me.

I noticed a bug, however.

Take the following statement:

	CREATE PROCEDURE Example_Sproc
		@SomeValue BIT,
		@SomeOtherValue SMALLINT,
		@Days INT
	AS
	BEGIN
		SET NOCOUNT ON;
 
		INSERT INTO Example (
 			SomeValue,
			SomeOtherValue,
			Days
		)
		OUTPUT inserted.Id
		VALUES (
 			@SomeValue,
			@SomeOtherValue,
			@Days
		);
	 
	END

This is a perfectly valid TSQL statement and compiles/executes. It's upset in particular about the column name `Days`.

It causes ANTLR4 to produce parser errors. In particular:

	errorListener.Errors	Count = 3	System.Collections.Generic.List<string>
	[0]	"missing END at 'INSERT'"	string
	[1]	"mismatched input 'Days' expecting {CALLED, DATA_COMPRESSION, EVENTDATA, FILLFACTOR, FORCESEEK, OFFSETS, PAGE, PUBLIC, RAW, RETURN, RETURNS, ROWCOUNT, SERVER, SOURCE, TARGET, ABSOLUTE, APPLY, AUTO, AVG, CALLER, CAST, CATCH, CHECKSUM_AGG, COMMITTED, CONCAT, CONCAT_NULL_YIELDS_NULL, CONTROL, COOKIE, COUNT, COUNT_BIG, DELAY, DELETED, DENSE_RANK, DISABLE, DYNAMIC, ENCRYPTION, EXPAND, FAST, FAST_FORWARD, FIRST, FOLLOWING, FORCE, FORCED, FORWARD_ONLY, FULLSCAN, GLOBAL, GROUPING, GROUPING_ID, HASH, IGNORE_NONCLUSTERED_COLUMNSTORE_INDEX, IMPERSONATE, INPUT, INSENSITIVE, INSERTED, ISOLATION, KEEP, KEEPFIXED, KEYSET, LAST, LEVEL, LOCAL, LOCK_ESCALATION, LOGIN, LOOP, MARK, MAX, MAXDOP, MAXRECURSION, MIN, MODIFY, NAME, NEXT, NOCOUNT, NOEXPAND, NORECOMPUTE, NTILE, NUMBER, OFFSET, ONLINE, ONLY, OPTIMISTIC, OPTIMIZE, OUT, OUTPUT, OWNER, PARAMETERIZATION, PARTITION, PATH, PRECEDING, PRIOR, PRIVILEGES, RANGE, RANK, READONLY, READ_ONLY, RECOMPILE, RELATIVE, REMOTE, REPEATABLE, ROBUST, ROOT, ROW, ROWGUID, ROWS, ROW_NUMBER, SAMPLE, SCHEMABINDING, SCROLL, SCROLL_LOCKS, SELF, SERIALIZABLE, SIMPLE, SIZE, SNAPSHOT, SPATIAL_WINDOW_MAX_CELLS, STATIC, STATS_STREAM, STDEV, STDEVP, SUM, TEXTIMAGE_ON, THROW, TIES, TIME, TRY, TYPE, TYPE_WARNING, UNBOUNDED, UNCOMMITTED, UNKNOWN, USING, VAR, VARP, VIEWS, VIEW_METADATA, WORK, XML, XMLNAMESPACES, DOUBLE_QUOTE_ID, SQUARE_BRACKET_ID, ID}"	string
	[2]	"mismatched input '<EOF>' expecting CONVERSATION"	string

Wrap `Days` in square brackets, and the TSQL string parses with success.

Not a huge issue at all, just thought I'd highlight it.

**Comment by tuxmania87** [comment]
Did you try to put Days in Quotes? Like that:

```
CREATE PROCEDURE Example_Sproc
	@SomeValue BIT,
	@SomeOtherValue SMALLINT,
	@Days INT
AS
BEGIN
	SET NOCOUNT ON;

	INSERT INTO Example (
		SomeValue,
		SomeOtherValue,
		[Days]
	)
	OUTPUT inserted.Id
	VALUES (
		@SomeValue,
		@SomeOtherValue,
		@Days
	);
 
END
```

**Comment by teverett** [comment]
This example, on the source tree, exhibits the failed behaviour:

https://github.com/antlr/grammars-v4/blob/master/tsql/examples/pr754.sql

**Comment by vonloxley** [comment]
This would be fixed by #945.
